transitions/utils.js

```diff
@@ -26,5 +32,32 @@ module.exports = {
       messages.addMessage(default_msg);
       messages.addError(default_msg.doc, default_msg.message);
     }
+  },
+  addUniqueId: function(db, doc, callback) {
+    var potentialIds = _.map(Array(IDS_TO_GENERATE), _.partial(ids.generate, currentIdLength));
+
+    utils.getRegistrations({
+        db: db,
+        ids: potentialIds
+    }, function(err, registrations) {
+        if (err) {
+            return callback(err);
+        }
+
+        var uniqueIds = _.reject(potentialIds, function(id) {
+          return _.find(registrations, function(registration) {
+            return registration.key === id;
```

Comment:
Little worried about matching if our `patient_id` field is a string or a number here...